src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -155,36 +158,35 @@ def on_revoke(consumer: Consumer, partitions: List[TopicPartition]) -> None:
 
         self.consumer.subscribe([self.topic], on_assign=on_assign, on_revoke=on_revoke)
 
-        try:
-            i = 0
-            while True:
-                message = self.consumer.poll(0.1)
-                if message is None:
-                    continue
+        i = 0
+        while not self.__shutdown_requested:
+            message = self.consumer.poll(0.1)
+            if message is None:
+                continue
 
-                error = message.error()
-                if error is not None:
-                    raise KafkaException(error)
+            error = message.error()
+            if error is not None:
+                raise KafkaException(error)
 
-                i = i + 1
+            i = i + 1
 
-                with sentry_sdk.start_transaction(
-                    op="handle_message",
-                    name="query_subscription_consumer_process_message",
-                    sampled=True,
-                ), metrics.timer("snuba_query_subscriber.handle_message"):
-                    self.handle_message(message)
+            with sentry_sdk.start_transaction(
+                op="handle_message",
+                name="query_subscription_consumer_process_message",
+                sampled=True,
+            ), metrics.timer("snuba_query_subscriber.handle_message"):
+                self.handle_message(message)
 
-                # Track latest completed message here, for use in `shutdown` handler.
-                self.offsets[message.partition()] = message.offset() + 1
+            # Track latest completed message here, for use in `shutdown` handler.
+            self.offsets[message.partition()] = message.offset() + 1
 
-                if i % self.commit_batch_size == 0:
-                    logger.debug("Committing offsets")
-                    self.commit_offsets()
-        except KeyboardInterrupt:
-            pass
```

Comment:
https://github.com/getsentry/sentry/pull/24316#issuecomment-793145340 was accurate, this never worked correctly outside of tests.

```
ted@veneno % sentry run query-subscription-consumer                                                                                                                                                                                                                                                                                                   [0] ~/workspaces/sentry/sentry
23:23:55 [WARNING] sentry.utils.geo: settings.GEOIP_PATH_MMDB not configured.
23:23:59 [INFO] sentry.plugins.github: apps-not-configured
/Users/ted/workspaces/sentry/sentry/src/sentry/utils/batching_kafka_consumer.py:51: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
  logger.warn("Topic '%s' or its partitions are not ready, retrying...", topic)
23:24:00 [WARNING] batching-kafka-consumer: Topic 'events-subscription-results' or its partitions are not ready, retrying...
23:24:03 [INFO] sentry.snuba.query_subscription_consumer: query-subscription-consumer.on_assign (offsets='{0: None}' partitions='[TopicPartition{topic=events-subscription-results,partition=0,offset=-1001,error=None}]')
^C23:24:03 [INFO] sentry.snuba.query_subscription_consumer: query-subscription-consumer.commit_offsets (offsets='{0: None}' partitions='None')
23:24:03 [INFO] sentry.snuba.query_subscription_consumer: query-subscription-consumer.commit_offsets (offsets='{0: None}' partitions='[0]')
23:24:03 [INFO] sentry.snuba.query_subscription_consumer: query-subscription-consumer.on_revoke (offsets='{}' partitions='[TopicPartition{topic=events-subscription-results,partition=0,offset=-1001,error=None}]')
Traceback (most recent call last):
  File "/Users/ted/workspaces/sentry/sentry/.venv/bin/sentry", line 33, in <module>
    sys.exit(load_entry_point('sentry', 'console_scripts', 'sentry')())
  File "/Users/ted/workspaces/sentry/sentry/src/sentry/runner/__init__.py", line 164, in main
    cli(prog_name=get_prog(), obj={}, max_content_width=100)
  File "/Users/ted/workspaces/sentry/sentry/.venv/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/ted/workspaces/sentry/sentry/.venv/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/ted/workspaces/sentry/sentry/.venv/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/ted/workspaces/sentry/sentry/.venv/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/ted/workspaces/sentry/sentry/.venv/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/ted/workspaces/sentry/sentry/.venv/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/ted/workspaces/sentry/sentry/.venv/lib/python3.6/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/ted/workspaces/sentry/sentry/src/sentry/runner/decorators.py", line 66, in inner
    return ctx.invoke(f, *args, **kwargs)
  File "/Users/ted/workspaces/sentry/sentry/.venv/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/ted/workspaces/sentry/sentry/.venv/lib/python3.6/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/ted/workspaces/sentry/sentry/src/sentry/runner/decorators.py", line 28, in inner
    return ctx.invoke(f, *args, **kwargs)
  File "/Users/ted/workspaces/sentry/sentry/.venv/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/ted/workspaces/sentry/sentry/src/sentry/runner/commands/run.py", line 387, in query_subscription_consumer
    subscriber.run()
  File "/Users/ted/workspaces/sentry/sentry/src/sentry/snuba/query_subscription_consumer.py", line 161, in run
    message = self.consumer.poll(0.1)
RuntimeError: Consumer closed
sentry run query-subscription-consumer  5.06s user 0.83s system 48% cpu 12.177 total
```